st3/sublime_lib/syntax.py

```diff
@@ -52,14 +53,14 @@ def get_xml_metadata(text):
 
 
 def get_syntax_metadata(path):
-    if path.endswith('.sublime-syntax'):
-        meta = get_yaml_metadata(sublime.load_resource(path))
-    elif path.endswith('.tmLanguage'):
-        meta = get_xml_metadata(sublime.load_binary_resource(path))
+    if path.suffix == '.sublime-syntax':
+        meta = get_yaml_metadata(path.read_text())
+    elif path.suffix == '.tmLanguage':
+        meta = get_xml_metadata(path.read_bytes())
     else:
         raise TypeError("%s is not a syntax definition." % path)
 
-    return SyntaxInfo(path=path, **meta)
+    return SyntaxInfo(path=str(path), **meta)
```

Comment:
Would be useful if we could just use the `ResourcePath` here, but that'd be breaking. :cry:

Reply:
Same.